Sprint-3/alarmclock/alarmclock.js

```diff
@@ -1,4 +1,63 @@
-function setAlarm() {}
+let countdownTimer;
+let flashInterval;
+
+function setAlarm() {
+  document.getElementById("stop").addEventListener("click", () => {
+    if (flashInterval) {
+      clearInterval(flashInterval);
+      flashInterval = null;
+      document.body.style.backgroundColor = "";
+    }
+  });
+
+  let alarmTime = Number(document.querySelector("input").value);
+
+  if (!alarmTime || isNaN(alarmTime) || alarmTime <= 0) {
+    alert("Please enter a valid positive number for the alarm time");
+    return;
+  }
+
+  let timeLeft = alarmTime;
+
+  function formatTime(time) {
+    let minutes = Math.floor(time / 60);
+    let seconds = time % 60;
+    return `${minutes.toString().padStart(2, "0")}:${seconds
+      .toString()
+      .padStart(2, "0")}`;
+  }
+
+  function updateTitle(time) {
+    document.getElementById(
+      "timeRemaining"
+    ).innerHTML = `Time Remaining: ${formatTime(time)}`;
```

Comment:
Are you including any HTML in the generated text here? If not, could you use `innerText` instead? Have a read of https://owasp.org/www-community/attacks/xss/ to see why we may prefer `innerText`

Reply:
@illicitonion Thanks for the suggestion. Since the content is plain text without any HTML, I’ve switched from innerHTML to textContent to ensure safer rendering and avoid any XSS vulnerabilities.